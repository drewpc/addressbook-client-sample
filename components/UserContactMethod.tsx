import parsePhoneNumber from 'libphonenumber-js';

function DisplayEmail(props: {contact: ApiEmailContactMethod}): JSX.Element {
    const {contact} = props;
    return <li>[email] {contact.label} - &nbsp;
        <a href={"mailto:" + contact.address}>{contact.address}</a>
    </li>
}

function DisplayPhone(props: {contact: ApiPhoneContactMethod}): JSX.Element {
    const {contact} = props;
    const phoneNumber = parsePhoneNumber(`+${contact.country_code}${contact.address}`);

    return <li>[phone] {contact.label} - &nbsp;
        <a href={phoneNumber.getURI()}>{phoneNumber.formatInternational()}</a>
    </li>
}

function DisplaySms(props: {contact: ApiSmsContactMethod}): JSX.Element {
    const {contact} = props;
    const phoneNumber = parsePhoneNumber(`+${contact.country_code}${contact.address}`);
    return <li>[sms] {contact.label} - {phoneNumber.formatInternational()}
    </li>
}

export default function UserContactMethod(props: {contact: ApiContactMethods}): JSX.Element {
    const {contact} = props;

    // Display phone, sms, or email contact information only.

    let displayValue;
    switch(contact.type) {
        case "phone_contact_method":
            displayValue = <DisplayPhone contact={contact} />
            break;

        case "sms_contact_method":
            displayValue = <DisplaySms contact={contact} />
            break;

        case "email_contact_method":
            displayValue = <DisplayEmail contact={contact} />
            break;

        default:
            displayValue = <></>;
    }

    return displayValue;
}