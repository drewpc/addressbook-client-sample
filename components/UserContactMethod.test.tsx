import React from 'react';
import renderer from 'react-test-renderer';
import UserContactMethod from './UserContactMethod';

const emailContact = {
    id: "a",
    type: "email_contact_method",
    label: "Home",
    address: "user@email.com",
} as ApiEmailContactMethod;

const phoneContact = {
    id: "b",
    type: "phone_contact_method",
    label: "Mobile",
    country_code: 1,
    address: "7205551212",
} as ApiPhoneContactMethod;

const smsContact = {
    id: "c",
    type: "sms_contact_method",
    label: "Mobile",
    country_code: 1,
    address: "7205551212",
} as ApiSmsContactMethod;

const pushContact = {
    type: "push_notification_contact_method",
    id: "d",
} as ApiPushNotificationContactMethod;

describe('User Contact Method component', () => {
    it('renders a phone contact', () => {
        const tree = renderer
            .create(<UserContactMethod contact={phoneContact}/>)
            .toJSON();
        expect(tree).toMatchSnapshot();
    });

    it('renders an email contact', () => {
        const tree = renderer
            .create(<UserContactMethod contact={emailContact}/>)
            .toJSON();
        expect(tree).toMatchSnapshot();
    });

    it('renders an sms contact', () => {
        const tree = renderer
            .create(<UserContactMethod contact={smsContact}/>)
            .toJSON();
        expect(tree).toMatchSnapshot();
    });

    it('does not render a push notification contact', () => {
        const tree = renderer
            .create(<UserContactMethod contact={pushContact}/>)
            .toJSON();
        expect(tree).toMatchSnapshot();
    });
});
