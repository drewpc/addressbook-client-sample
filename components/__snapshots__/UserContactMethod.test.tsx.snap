// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`User Contact Method component does not render a push notification contact 1`] = `null`;

exports[`User Contact Method component renders a phone contact 1`] = `
<li>
  [phone] 
  Mobile
   -  
  <a
    href="tel:+17205551212"
  >
    +1 720 555 1212
  </a>
</li>
`;

exports[`User Contact Method component renders an email contact 1`] = `
<li>
  [email] 
  Home
   -  
  <a
    href="mailto:user@email.com"
  >
    user@email.com
  </a>
</li>
`;

exports[`User Contact Method component renders an sms contact 1`] = `
<li>
  [sms] 
  Mobile
   - 
  +1 720 555 1212
</li>
`;
